src/dds/build/plan/compile_exec.cpp

```diff
@@ -168,6 +172,11 @@ handle_compilation(const compile_ticket& compile, build_env_ref env, compile_cou
          */
     }
 
+    if (compiled_okay && compile.command.touch_path) {
+        dds_log(trace, "Touching specified file: {}", compile.command.touch_path->string());
+        std::ofstream(*compile.command.touch_path);  // Touch the file
+    }
```

Comment:
`dds` doesn't use mtime comparisons between input/output files to perform incremental compilation, so no output file is needed for a syntax-only check.

`dds` records the mtime of input files in the db and then compares the mtime of those same files when it executes again. It also checks if the output object file exists, but since there is no object file path associated with a syntax-only file, this check should be skipped. Refer to `mk_compile_ticket` in `compile_exec.cpp`.